skimage/morphology/_near_objects_cy.pyx

```diff
@@ -0,0 +1,168 @@
+#cython: cdivision=True
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+
+
+"""Cython code used in `remove_near_objects` function."""
+
+
+import numpy as np
+cimport numpy as cnp
+
+
+# Must be defined to use QueueWithHistory
+ctypedef Py_ssize_t QueueItem
+
+
+include "_queue_with_history.pxi"
+
+
+ctypedef fused dtype_t:
```

Comment:
The fused type `np_real_numeric` (already defined in ``skimage._shared.fused_numerics.pxd) can be used here :)

Reply:
Oh, nice. Didn't know we had that!